internal/upload/album_test.go

```diff
@@ -98,3 +118,121 @@ func TestAlbumNameUsingFolderName(t *testing.T) {
 		assert.Equal(t, tt.out, albumNameUsingFolderName(tt.in))
 	}
 }
+
+func TestGetTemplateFunctionName(t *testing.T) {
+	var testData = []struct {
+		in  string
+		out string
+	}{
+		{in: "", out: ""},
+		{in: "$cutLeft(", out: "cutLeft"},
+		{in: "$cutLeft (", out: ""},
+		{in: "cutLeft(", out: ""},
+		{in: "$cutLeft(anything", out: "cutLeft"},
+	}
+
+	for _, tt := range testData {
+		assert.Equal(t, tt.out, getTemplateFunctionName(tt.in))
+	}
+}
+
+func TestParseAlbumNameTampleWithTokens(t *testing.T) {
+	timeObj := time.Date(2034, time.December, 31, 16, 5, 59, 0, time.UTC)
+	filePath := "/foo/bar/file.jpg"
+
+	var testData = []struct {
+		in  string
+		out string
+	}{
+		{in: "%_year%", out: "2034"},
+		{in: "%_day%", out: "31"},
+		{in: "%_month%", out: "12"},
+
+		{in: "%_folderpath%", out: "foo_bar"},
+		{in: "%_directory%", out: "bar"},
+		{in: "%_parent_directory%", out: "foo"},
+
+		{in: "%_time%", out: "16:05:59"},
+		{in: "%_time_en%", out: "04:05:59 PM"},
+		{in: "%_year%_%_year%", out: "2034_2034"},
+	}
+
+	for _, tt := range testData {
+		val, err := parseAlbumNameTemplate(tt.in, filePath, timeObj)
+		if err != nil {
+			t.Errorf("Unexpected error: %v", err)
+		}
+
+		assert.Equal(t, tt.out, val)
+	}
+}
+
+func TestParseAlbumNameTampleWithFunctions(t *testing.T) {
```

Comment:
Same as above, I'm wondering if this `Tample` is a short name... in that case, feel free to leave is as it is